README

```diff
@@ -1241,6 +1249,30 @@ Various utility functions ontop of @GEB-CATEGORIES
 
     Grabs the codomain of the morphism
 
+- [generic-function] CURRY F
+
+    Curries the given object, returns a <SUBSTMORPH>
+    
+    The <SUBSTMORPH> given must have its DOM be of a PROD type, as CURRY
+    invokes the idea of
+    
+    if f : (PROD a b) → c
+    
+    for all `a`, `b`, and `c` being an element of <SUBSTMORPH>
+    
+    then: (curry f): a → c^b
+    
+    where c^b means c to the exponent of b (EXPT c b)
+    
+    ```
+    Γ, f : a × b → c,
+    --------------------
+    (curry f) : a → c^b
+    ```
+    
+    In category terms, `a → c^b` is isomorphic to `a → b → c`
```

Comment:
Again, might be a good idea to change up the notation a bit

Reply:
perhaps, though I think here this gives a more intuitive feeling for it?

We should have a followup pr on this issue as well